packages/cactus-plugin-bungee/src/main/typescript/web-services/client-endpoint.ts

```diff
@@ -0,0 +1,103 @@
+import type { Express, Request, Response } from "express";
+
+import {
+  IWebServiceEndpoint,
+  IExpressRequestHandler,
+  IEndpointAuthzOptions,
+} from "@hyperledger/cactus-core-api/";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+  IAsyncProvider,
+} from "@hyperledger/cactus-common";
+
+import { registerWebServiceEndpoint } from "@hyperledger/cactus-core";
+
+import OAS from "../../json/openapi.json";
+import { PluginBUNGEE } from "../plugin-bungee";
+
+export interface CreateViewEndpointOptions {
+  logLevel?: LogLevelDesc;
+  bungee: PluginBUNGEE;
+}
+
+export class ClientEndpointV1 implements IWebServiceEndpoint {
+  public static readonly CLASS_NAME = "ClientEndpointV1";
+
+  private readonly log: Logger;
+
+  public get className(): string {
+    return ClientEndpointV1.CLASS_NAME;
+  }
+
+  constructor(public readonly options: CreateViewEndpointOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    Checks.truthy(options.bungee, `${fnTag} arg options.connector`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+  
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    return this.handleRequest.bind(this);
+  }
+
+  public getPath(): string {
+    const apiPath =
+      OAS.paths[
+        "/api/v1/plugins/@hyperledger/cactus-plugin-bungee/create-view"
+      ];
+    return apiPath.post["x-hyperledger-cacti"].http.path;
+  }
+
+  public getVerbLowerCase(): string {
+    const apiPath =
+      OAS.paths[
+        "/api/v1/plugins/@hyperledger/cactus-plugin-bungee/create-view"
+      ];
+    return apiPath.post["x-hyperledger-cacti"].http.verbLowerCase;
+  }
+
+  public getOperationId(): string {
+    return OAS.paths[
+      "/api/v1/plugins/@hyperledger/cactus-plugin-bungee/create-view"
+    ].post.operationId;
+  }
+
+  getAuthorizationOptionsProvider(): IAsyncProvider<IEndpointAuthzOptions> {
+    // TODO: make this an injectable dependency in the constructor
+    return {
+      get: async () => ({
+        isProtected: true,
+        requiredRoles: [],
+      }),
+    };
+  }
+
+  public async registerExpress(
+    expressApp: Express,
+  ): Promise<IWebServiceEndpoint> {
+    await registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  public async handleRequest(req: Request, res: Response): Promise<void> {
+    const reqTag = `${this.getVerbLowerCase()} - ${this.getPath()}`;
+    this.log.debug(reqTag);
+    try {
+      const view = await this.options.bungee.onCreateView(req.body);
+      res.status(200).json(view);
+    } catch (ex) {
+      this.log.error(`Crash while serving ${reqTag}`, ex);
+      res.status(500).json({
+        message: "Internal Server Error",
+        error: ex?.stack || ex?.message,
+      });
+    }
```

Comment:
@eduv09 Please refactor the error handling to the new pattern we've (recently) established so that it is more robust and handles HTTP specific error codes automatically:

You can see an example (that is pretty much one to one copy-pastable when it comes to the `catch` block) here in this file on the main branch:

`packages/cactus-core/src/main/typescript/web-services/get-open-api-spec-v1-endpoint-base.ts`